src/json/pull_parser.cr

```diff
@@ -250,35 +250,35 @@ class JSON::PullParser
   end
 
   def read?(klass : Int8.class)
-    read_int.to_i8 if kind == :int
+    read_int.to_i8! if kind == :int
```

Comment:
Shouldn't this, and all of the others, raise on overflow? Or maybe not raise, but return `nil` because this is named `read?`

Reply:
I would answer all json/yaml comments with: the change is to keep the current behavior. And can't be refactored until overflow is turned on by default.

Reply:
Makes total sense, thanks!